### PR TITLE
/endpoint-history: Support PostgreSQL

### PR DESCRIPTION
**Pull request by synackd**
Introduces one of multiple fixes for #20.

Modify the `/endpoint-history` endpoint to support PostgreSQL, not just EtcD.

### CODE REVIEW

File: internal/postgres/common.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/jmoiron/sqlx/reflectx"
+)
+
+const (
+	xNameRegex = `^x([0-9]{1,4})c([0-7])(s([0-9]{1,4}))?b([0])(n([0-9]{1,4}))?$`
```
Comment (davidallendj) [inline internal/postgres/common.go:26]: Is this used for checking of a valid xname? If so, is it possible to use the [`xname`](https://github.com/Cray-HPE/hms-xname) library instead? I'm not sure if we'd actually want to go this route, but I do find it interesting that this is here.

Comment (davidallendj) [reply internal/postgres/common.go:26]: After a closer look, it seems like using that library might not be ideal for this situation.

Comment (synackd) [reply internal/postgres/common.go]: Fixed in 2c58ccf.

Yeah, that makes more sense than using the basic regular expression I had...

Comment (synackd) [reply internal/postgres/common.go]: I think using the library makes more sense until either Ochami gets a fork of the xnames repository or xnames are dropped. Whichever comes first.

**Review by davidallendj — APPROVED** [review]
LGTM and it didn't crash, so I'd say it's working.

**Comment by synackd** [comment]
Thanks @davidallendj. I've done some preliminary testing on two clusters (including booting nodes and querying the endpoint) and it works as expected.

**Pull request merged**